src/databricks/labs/ucx/framework/crawlers.py

```diff
@@ -233,3 +234,30 @@ def _snapshot(self, fetcher, loader) -> list[any]:
     def _append_records(self, items):
         logger.debug(f"[{self._full_name}] found {len(items)} new records for {self._table}")
         self._backend.save_table(self._full_name, items, self._klass, mode="append")
+
+
+class SchemaDeployer:
+    def __init__(self, sql_backend: SqlBackend, inventory_schema: str, mod: any):
+        self._sql_backend = sql_backend
+        self._inventory_schema = inventory_schema
+        self._module = mod
+
+    def deploy_schema(self):
+        logger.info(f"Ensuring {self._inventory_schema} database exists")
+        self._sql_backend.execute(f"CREATE SCHEMA IF NOT EXISTS hive_metastore.{self._inventory_schema}")
+
+    def deploy_table(self, name: str, klass: type):
+        logger.info(f"Ensuring {self._inventory_schema}.{name} table exists")
+        self._sql_backend.create_table(f"hive_metastore.{self._inventory_schema}.{name}", klass)
```

Comment:
It's a good point. I don't know if re-creating is better than removing all records - in case of removal, we can do a rollback to a previous delta table version. We already have an issue for it: https://github.com/databrickslabs/ucx/issues/224

Reply:
I would say a insert-overwrite + schema evolution to handle version increments of the product without manual intervention.
Let's keep this thing out of the scope of this PR for now ;)